### PR TITLE
Move import contributors to a custom question type

### PR DESCRIPTION
**Pull request by atelic**
Per https://github.com/barbour-em/osf.io/commit/af7b10d5f40cea9a08907889fadce629e3b69ade#commitcomment-14336153

### CODE REVIEW

File: website/static/js/registrationEditorExtensions.js [hunk 1]
```diff
@@ -142,7 +142,19 @@ var Uploader = function(data) {
     $.extend(self, data);
 };
 
+var AuthorImport = function(data, $root) {
+    var self = this;
+
+    self._orig = data;
+    self.contributors = $root.contributors();
+    self.currentQuestion = $root.currentQuestion;
+    self.currentQuestion($root.find('authorship'));
```
Comment (samchrisinger) [inline website/static/js/registrationEditorExtensions.js:152]: Can you explain the need for modifying `currentQuestion`?

File: website/static/js/registrationUtils.js [hunk 2]
```diff
@@ -1139,7 +1140,10 @@ RegistrationEditor.prototype.authorDialog = function() {
                         }
                     });
                     if ( authors ) {
-                        self.currentQuestion().setValue(authors.join(', '));
+                        $.each(self.currentQuestion().properties, function(name, subQuestion){
```
Comment (samchrisinger) [inline website/static/js/registrationUtils.js]: This logic (`authorDialog`) should all get encapsulated in your extension code.

**Comment by samchrisinger** [comment]
I'd like to see the `authorDialog` code moved to the extensions file. This is not mission-critical, but it keeps the abstraction cleaner and more separate. 

Also, rather than looking for a certain subquestion name, can we use subquestion type?

File: website/static/js/registrationEditorExtensions.js (hunk 1 above)
Comment (atelic) [inline website/static/js/registrationEditorExtensions.js:152]: The Import Contributors button could be pressed from anywhere on the page but the `authorsDialog` uses `currentQuestion.setValue` to fill out the field.

**Comment by atelic** [comment]
`authorDialog` has been moved.

The problem is that the subquestion type would just be "string" or "textarea" like any other question

**Comment by samchrisinger** [comment]
I guess I was imagining:

``` json
  "properties": {
                "authorship": {
                    "id": "q3a",
                    "type": "string",
                    "format": "textarea",
                    "required": true
                },
                "importAuthors": {
                    "id": "q3b",
                    "type": "osf-author-import",
                    "format": "osf-import-button"
                }
            }
```

would look more like:

``` json
"properties": {
    "authorship": {
         "id": "q3",
        "type": "osf-author-import",
        "required": true
    }
}
```

File: website/static/js/registrationEditorExtensions.js (hunk 1 above)
Comment (samchrisinger) [inline website/static/js/registrationEditorExtensions.js:152]: In this context though the `data` argument refers to the context of the 'authorship' question. I don't think there's necessarily a problem with modifying the root's `currentQuestion`, but it seems unnecessary given the information provided via the existing context.

**Comment by samchrisinger** [comment]
@barbour-em thank you for the work here. I'm merging manually and will resolve any concerns during merge.

**Pull request closed without merging**